chemprop/cli/train.py

```diff
@@ -584,6 +585,9 @@ def main(args):
             for i in range(n_components)
         ]
         if args.mpn_shared:
+            if has_mol_and_rxn:
```

Comment:
You define has_mol_and_rxn above, but only use it once. I think it would be better to combine them and simplify into `if args.reaction_columns is not None and args.smiles_columns is not None:`

Reply:
Thanks! Done